src/storage/mod.rs

```diff
@@ -104,13 +104,28 @@ impl Mutation {
     }
 }
 
+/// Represents the status of a transaction.
+#[derive(PartialEq, Debug)]
+pub enum TxnStatus {
+    /// The txn is just rolled back by the current command.
+    Rollbacked,
+    /// The txn was already rolled back before.
+    RollbackedBefore,
+    /// The txn doesn't exist.
+    NotExist,
```

Comment:
Perhaps it's unnecessary and incorrect to distinguish `NotExist` with `RollbackedBefore`? A collapsed rollback may be incorrectly treated as `NotExist` .

Reply:
I prefer keep it for now. The implementation may change according to the discussion in our slack channel.

Reply:
Understand it. There are quite a few tricky points in our transaction model. We can think later if these values can always have correct meanings...